src/PetriEngine/Colored/Reduction/RedRulePreAgglomeration.cpp

```diff
@@ -0,0 +1,213 @@
+                }
+
+                // Update
+                for (const auto& prod : originalProducers){
+                    const Transition &producer = red.transitions()[prod];
+//                    uint32_t k = 1;
+//                    if (!kIsAlwaysOne[n]){
+//                        k = red.getOutArc(producer, pid)->expr / w;
+//                    }
+
+                    // One for each number of firings of consumer possible after one firing of producer
+//                    for (uint32_t k_i = 1; k_i <= k; k_i++){
+                        // Create new transition with effect of firing the producer, and then the consumer k_i times
+                        auto id = red.newTransition(nullptr);
+
+                        // Re-fetch the transition pointers as it might be invalidated, I think that's the issue?
+                        const Transition &producerPrime = red.transitions()[prod];
+                        const Transition &consumerPrime = red.transitions()[originalConsumers[n]];
+                        const Transition& newtran = red.transitions()[id];
+
+                        // Arcs from consumer
+                        for (const auto& arc : consumerPrime.output_arcs) {
+                            ArcExpression_ptr expr = arc.expr;
+                            red.addOutputArc(newtran, red.places()[arc.place], expr);
+                        }
+                        for (const auto& arc : consumerPrime.input_arcs){
+                            if (arc.place != pid){
+                                ArcExpression_ptr expr = arc.expr;
+                                red.addInputArc(red.places()[arc.place], newtran, expr, arc.inhib_weight);
+                            }
+                        }
+
+                        for (const auto& arc : producerPrime.input_arcs){
+                            ArcExpression_ptr expr = arc.expr;
+                            red.addInputArc(red.places()[arc.place], newtran, expr, arc.inhib_weight);
+                        }
+
+//                        if (k_i != k){
+//                            ArcExpression_ptr k_expr = (k-k_i)*w;
+//                            red.addOutputArc(newtran, place, k_expr);
+//                        }
+//                    }
+                }
+                red.skipTransition(originalConsumers[n]);
+                continueReductions = true;
+                _applications++;
+            }
+
+            if (place._post.empty()) {
+                if (!preserveStutter){
+                    // The producers of place will become purely consuming transitions when it is gone, which can sometimes be removed
+                    // The places they consume from aren't allowed to be in the query, but if they were we couldn't reach this point either.
+                    auto transitions = place._pre;
+                    for (auto tran_id : transitions)
+                        red.skipTransition(tran_id);
+                }
```

Comment:
Agglomeration reduktioner er ikke valide for stutter sensitive queries, så du burde bail med det samme,  når `preserveStutter` er sand.